transport/transport.go

```diff
@@ -606,3 +612,30 @@ func wait(ctx context.Context, done, goAway, closing <-chan struct{}, proceed <-
 		return i, nil
 	}
 }
+
+const (
+	// DefaultTOS is the default value for TOS
+	DefaultTOS = 0
+)
+
+// Default implementation to fallback to when user doesn't provide getTos mechanism
+// getTOS retrieves tos value for a connection
+// If an error occurs it returns the default value for tos
+func defaultGetTOS(conn net.Conn) int {
+	var fd int
+	switch conn := conn.(type) {
+	case *net.TCPConn, *net.UDPConn, *net.UnixConn:
+		fd = int(reflect.ValueOf(conn).Elem().FieldByName("conn").FieldByName("fd").Elem().FieldByName("sysfd").Int())
```

Comment:
You're kidding me... we can't do better than this?

At the very least you should link to https://github.com/golang/go/issues/9661 , but...